src/corecel/sys/Device.hh

```diff
@@ -105,8 +114,8 @@ class Device
     //! Number of execution units per compute unit (1 for NVIDIA, 4 for AMD)
     unsigned int eu_per_cu() const { return eu_per_cu_; }
 
-    //! Default number of threads per block
-    unsigned int default_block_size() const { return default_block_size_; }
+    //! Number of threads per block
+    unsigned int block_size() const { return block_size_; }
```

Comment:
This parameter was meant to be "for kernels that don't specifically override the block size"  rather than "if the user doesn't specify the block size". Since the `constexpr` version is (I think?) limiting the compile-time "max threads per block", would it be reasonable to rename that one to `max_block_size` and restoring this variable name?

Reply:
Actually, I'm not convinced about `max_block_size` I don't think any value make sense for it as mentioned in my previous comment.

Reply:
Well, doesn't it reduce the local memory overflow if we constrain the max block size to the "maximum block size we will ever use"? All of our transport-loop kernels use a block size of 256, but for a couple of them the "maximum" thread size is higher than that. Won't it be better to have the compiled-in maximum be the in-practice maximum?

Reply:
Is "maximum block size we will ever use" 1024? If we set this as default and apply __launch_bounds__(1024) to kernels that we launch with a block size of 256 we still limit the number of registers/thread and increase local memory usage so I don't see a use case for a compile time max.

Reply:
Right now almost every kernel launches with `Device::default_block_size()` which defaults to 256. What I mean is that, for example, the pre-step kernel has these statics on summit (V100):
```json
{
"const_mem": 20592,
"heap_size": 8388608,
"local_mem": 0,
"max_blocks_per_cu": 3,
"max_threads_per_block": 896,
"max_warps_per_eu": 24,
"name": "pre-step",
"num_regs": 70,
"occupancy": 0.375,
"print_buffer_size": 5242880,
"stack_size": 8192,
"threads_per_block": 256
}
```
We only ever launch it with 256 (`threads_per_block`) but the automatically computed launch limits set 
`max_threads_per_block` of 896. If we reduce the maximum to `256` as well, then won't the occupancy go up (or the register usage be increased to reduce memory bandwidth?)

Reply:
I see, yes in that case using `__launch_bounds__` to tell the compiler that we'll never use more than 256 threads per block could improve occupancy / reduce local memory usage though we'd probably want to profile it to make sure. 

I don't think that properties max_threads_per_block does anything though, we just read the value from device properties and print it.